addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -0,0 +1,314 @@
+
+        output_head = '%s%s%s' % (
+            _csv_row(fk_head_list, delimiter),
+            _csv_row(lt_head_list, delimiter),
+            _csv_row(of_head_list, delimiter),
+        )
+
+        for move in self.filtered(lambda m: m.state == 'posted'):
+            eTax = move._prepare_etax()
+
+            nik = str(move.partner_id.l10n_id_nik) if not move.partner_id.l10n_id_npwp else ''
+
+            if move.l10n_id_replace_invoice_id:
+                number_ref = str(move.l10n_id_replace_invoice_id.name) + " replaced by " + str(move.name) + " " + nik
+            else:
+                number_ref = str(move.name) + " " + nik
+
+            street = ', '.join([x for x in (move.partner_id.street, move.partner_id.street2) if x])
+
+            invoice_npwp = '000000000000000'
+            if not move.partner_id.l10n_id_npwp:
+                if move.partner_id.l10n_id_npwp and len(move.partner_id.l10n_id_npwp) >= 12:
+                    invoice_npwp = move.partner_id.l10n_id_npwp
+                elif (not move.partner_id.l10n_id_npwp or len(move.partner_id.l10n_id_npwp) < 12) and move.partner_id.l10n_id_nik:
+                    invoice_npwp = move.partner_id.l10n_id_nik
+                invoice_npwp = invoice_npwp.replace('.', '').replace('-', '')
+
+            # Here all fields or columns based on eTax Invoice Third Party
+            eTax['KD_JENIS_TRANSAKSI'] = move.l10n_id_efaktur_id.name[0:2] if move.l10n_id_efaktur_id else 0
+            eTax['FG_PENGGANTI'] = move.l10n_id_efaktur_id.name[2] if move.l10n_id_efaktur_id else 0
+            eTax['NOMOR_FAKTUR'] = move.l10n_id_efaktur_id.name[3:] if move.l10n_id_efaktur_id else 0
+            eTax['MASA_PAJAK'] = move.invoice_date.month
+            eTax['TAHUN_PAJAK'] = move.invoice_date.year
+            eTax['TANGGAL_FAKTUR'] = '{0}/{1}/{2}'.format(move.invoice_date.day, move.invoice_date.month, move.invoice_date.year)
+            eTax['NPWP'] = invoice_npwp
+            eTax['NAMA'] = move.partner_id.name if eTax['NPWP'] == '000000000000000' else move.partner_id.l10n_id_tax_name or move.partner_id.name
+            eTax['ALAMAT_LENGKAP'] = move.partner_id.contact_address.replace('\n', '') if eTax['NPWP'] == '000000000000000' else move.partner_id.l10n_id_tax_address or street
+            eTax['JUMLAH_DPP'] = int(round(move.amount_untaxed, 0))
+            eTax['JUMLAH_PPN'] = int(round(move.amount_tax, 0))
+            eTax['ID_KETERANGAN_TAMBAHAN'] = '1' if move.l10n_id_kode_transaksi == '07' else ''
+            eTax['REFERENSI'] = number_ref
+
+            lines = move.line_ids.filtered(lambda x: x.product_id.id == int(dp_product_id) and x.price_unit < 0)
+            eTax['FG_UANG_MUKA'] = 0
+            eTax['UANG_MUKA_DPP'] = int(abs(sum(lines.mapped('price_subtotal'))))
+            eTax['UANG_MUKA_PPN'] = int(abs(sum(lines.mapped(lambda l: l.price_total - l.price_subtotal))))
```

Comment:
can not be derived from something else?

Reply:
It is the tax per invoice line. We already discussed this in other tasks...

Reply:
No, it depends on the product
```python
dp_product_id = self.env['ir.config_parameter'].sudo().get_param('sale.default_deposit_product_id')
...
lines = move.line_ids.filtered(lambda x: x.product_id.id == int(dp_product_id) and x.price_unit < 0)
```